### PR TITLE
Give examples for all cache configurations.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
#8373 documented how to setup Memcached for caching. The documentation is missing some information on a few caches. These are added in this PR.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: docs/sources/operations/caching.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ cache is configured to be in-memory by default.
 
 ## Before you begin
 
-- It is recommended to deploy four dedicated Memcached clusters.
+- It is recommended to deploy four or in the case of the Helm chart two dedicated Memcached clusters.
```
Comment (JStickler) [inline docs/sources/operations/caching.md:20]:
```suggestion
- It is recommended to deploy four, or in the case of the Helm chart two, dedicated Memcached clusters.
```

File: docs/sources/operations/caching.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ To enable and configure Memcached:
        ```
        --memory-limit=4096 --max-item-size=2m --conn-limit=1024
        ```
-    1. Query result cache
+    1. Query result, index queries and writes cache
```
Comment (JStickler) [inline docs/sources/operations/caching.md:34]: The instructions are a big vague here (and I know this is existing docs) and don't clearly specify where to make these configuration edits.  If this is three separate configuration files that need to be modified, we should list it as THREE separate steps, to make it absolutely clear to users that there are three separate places that this configuration needs to be applied.

File: docs/sources/operations/caching.md [hunk 3]
```diff
@@ -67,16 +67,24 @@ To enable and configure Memcached:
...
+                 host: <chunk cache memcached host>
                  service: <port name of memcached service>
+	   write_dedupe_cache_config: {
+             memcached:
+               batch_size: 100
+               parallelism: 100
+             memcached_client: 
+               host: <index writes memcached host>
+               service: <port name of memcached service>
+               consistent_hash: true
```
Comment (JStickler) [inline docs/sources/operations/caching.md:87]: Check the formatting on this example, in the preview the bit I've copy/pasted below isn't included in the code block.  

`write_dedupe_cache_config: { memcached: batch_size: 100 parallelism: 100 memcached_client: host: service: consistent_hash: true `

**Review by JStickler — COMMENTED** [review]
[docs team]  The example code in step 2 ii a needs to be fixed.  And then step 2 ii b should hopefully be numbered properly.

File: docs/sources/operations/caching.md (hunk 2 above)
Comment (jeschkies) [reply docs/sources/operations/caching.md:34]: Good point. The edits are in three sections of the same configuration file. I've tried to clarify this.

File: docs/sources/operations/caching.md [hunk 4]
```diff
@@ -66,17 +66,26 @@ To enable and configure Memcached:
              service: memcache
              default_validity: 12h
        ```
-    1. If the Loki configuration is used
-        1. Configure the chunk cache
+    1. If the Loki configuration is used modify the following three sections in
```
Comment (JStickler) [inline docs/sources/operations/caching.md:69]:
```suggestion
    1. If the Loki configuration is used, modify the following three sections in
```

**Review by JStickler — APPROVED** [review]
[docs team] LGTM

**Pull request merged**